webui-src/app/login.js

```diff
@@ -5,31 +5,45 @@ const displayErrorMessage = function(message) {
   m.render(document.getElementById('error'), message);
 };
 
-const verifyLogin = async function(uname, passwd, port) {
+const verifyLogin = async function(uname, passwd, port, server) {
   const loginHeader = {
     Authorization: 'Basic ' + btoa(uname + ':' + passwd)
   };
-  rs.setKeys('', '', port, false);
+  if (typeof server !== 'string' || server.indexOf(':') >= 0) {
+    displayErrorMessage("Server incorrect, please enter only ip or name");
+    return;
+  } else if (!server.trim()) {
+    displayErrorMessage("Server missing, please enter ip or name");
+    return;
+  }
+  var url = 'http://' + server.trim() + ":";
+  rs.setKeys('', '', port, false, url);
```

Comment:
How about instead of enforcing only ip/hostname, we just add `http` if it isn't present,
Something like:
```js
if (!server.startsWith('http://`) {
    // prepend http to string
}
```
What do you think?

Reply:
this way we fix it without notice to user (that i didn't like in first place). The empty check should be done anyway. Assuming someone adding another protocol (like https) maybe for tunneling reasons, it wouldn't work either.
The better assume should be to replace server and port with single url and let the user decide to override the complete endpoint.
Only if we see the file:// protocol, we ignore the current location. In this case, he run the file from disk (like described in README) and the protocol, host and port won't work.

Reply:
Makes sense, I guess if we are checking the URL, we should go all the way in and use regex. But we can do that later and doesn't have to do everything in this PR.